CMakeLists.txt

```diff
@@ -0,0 +1,294 @@
+    
+    list(APPEND bb_sources
+        src/b3/blake3_avx2.c
+        src/b3/blake3_avx512.c
+        src/b3/blake3_sse41.c
+    )
+
+    if(NOT MSVC)
+        list(APPEND bb_sources
+            src/b3/blake3_avx2_x86-64_unix.S
+            src/b3/blake3_avx512_x86-64_unix.S
+            src/b3/blake3_sse41_x86-64_unix.S
+        )
+    else()
+        # Disable blake3 convertion loss of data warnings 
+        set_source_files_properties( 
+            src/b3/blake3_avx2.c
+            src/b3/blake3_avx512.c
+            src/b3/blake3_sse41.c
+            PROPERTIES COMPILE_FLAGS
+            /wd4244
+        )
+    endif()
+elseif(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "arm64" OR ${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "aarch64")
+else()
+    message( FATAL_ERROR "Unsupported architecture '${CMAKE_HOST_SYSTEM_PROCESSOR}'" )
+endif()
+
+# OS
+if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+
+    file(GLOB_RECURSE src_linux RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} 
+        CONFIGURE_DEPENDS LIST_DIRECTORIES false
+        src/platform/unix/*.cpp
+        src/platform/linux/*.cpp
+    )
+    list(APPEND bb_sources ${src_linux})
+
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+    
+    file(GLOB_RECURSE src_win RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} 
+        CONFIGURE_DEPENDS LIST_DIRECTORIES false
+        src/platform/win32/*.cpp
+    )
+    list(APPEND bb_sources ${src_win})
+
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+
+    file(GLOB_RECURSE src_mac RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} 
+        CONFIGURE_DEPENDS LIST_DIRECTORIES false
+        src/platform/unix/*.cpp
+        src/platform/macos/*.cpp
+    )
+    list(APPEND bb_sources ${src_mac})
+
+else()
+    message( FATAL_ERROR "Unsupported operating system '${CMAKE_SYSTEM_NAME}'" )
+endif()
+
+# Dev config?
+file(GLOB_RECURSE src_dev RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} 
+    CONFIGURE_DEPENDS LIST_DIRECTORIES false
+    src/test/*.cpp
+)
+
+file(GLOB_RECURSE headers_dev RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} 
+    CONFIGURE_DEPENDS LIST_DIRECTORIES false
+    src/test/*.h
+)
+
+
+# Exe
+find_package(Threads REQUIRED)
+
+set(platform_libs)
+
+if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+    set(platform_libs numa)
+endif()
+
+set(bb_include_dirs 
+    ${INCLUDE_DIRECTORIES}
+    ${CMAKE_CURRENT_SOURCE_DIR}/src
+    ${bls_SOURCE_DIR}/include
+    ${relic_SOURCE_DIR}/include
+)
+
+
+# BladeBit
+add_executable(bladebit     src/main.cpp ${bb_sources} ${bb_headers})
+add_executable(bladebit_dev src/test/test_main.cpp ${bb_sources} ${src_dev} ${bb_headers} ${headers_dev})
+
+macro(config_proj tgt)
+
+    target_compile_options(${tgt} PRIVATE $<$<CONFIG:release>:${c_opts} ${release_c_opts}>)
+    target_compile_options(${tgt} PRIVATE $<$<CONFIG:debug>:${c_opts} ${debug_c_opts}>)
+    target_link_options(${tgt} PRIVATE $<$<CONFIG:release>:${link_opts} ${debug_link_opts}>)
+    target_link_options(${tgt} PRIVATE $<$<CONFIG:debug>:${link_opts} ${debug_link_opts}>)
```

Comment:
You're accidentally using `debug_link_opts` for both build types.

Also you may consider consolidating these using `$<if>` genex, as the available build types are binary here.

Reply:
Thanks for catching this!